Added MoveCommand and RotateCommand

ayusavin opened the pull request:
Added MoveCommand and RotateCommand, written all necessary interfaces, Commands fully covered by tests.

Commit:
Initialized servers source project and tests project, Dockefiles, GitHub Actions pipeline

Added Dockerfiles for building release project and running tests.

Commit:
Fixed typos in README and build.yml

Commit:
GitHub actions badge URI fixed

Commit:
Code coverage now in CI pipeline

Delete temporary evaulate_coverage.sh, updated Dockerfile.ci and build workflow

Commit:
Fix typos in build.yml

Commit:
Added current branch extraction in build.yml

Commit:
Automated publish: Sat Sep 10 18:01:10 UTC 2022 b2433ced1273a2cb15cb3bbed0e4e8e687efeacc

Commit:
Now coverage reports hosted on dedicated server

Updated Dockerfile.ci, now it generate report in .html, build.yml now send report on dedicated server. README.md coverage badges links on coverage reports.

Commit:
Merge branch 'dev' of github.com:Fantast03/ooaip2223 into dev

Commit:
fixed typo in build.yml

Commit:
Fixed typos in build.yml, Send coverage report step

Commit:
Fixed typos in build.yml, ITHUB_TOKEN now GITHUB_TOKEN

ayusavin committed:
Added move command, miscalaneous features

The project's sources have been moved to the src folder, added ICommand, IMovable interfaces, MoveCommand implementation of ICommand, with tests. Deleted main file and generation of the execution project file. Sending coverage report logic taken out to the .github/send_report.sh.

ayusavin committed:
Fixed build.yml branches typo

ayusavin committed:
Fixed permissions problem in build.yml, running send_report.sh

ayusavin committed:
Fixed typos in build.yml and send_report.sh

Commit:
Automated publish: Mon Oct 10 16:52:35 UTC 2022 f791d1a833c206a0ab71ede318dbcf753a0bea5d

ayusavin committed:
Fixed regular expression in the build.yml, added - and _ to process branches with this delimeter

ayusavin committed:
Merge branch 'feature-MoveCommand' of github.com:Fantast03/ooaip2223 into feature-MoveCommand

ayusavin committed:
BRANCH_NAME env variable finally returns to they place in the edit README.md statement in build.yml :)

ayusavin committed:
Fixed regex in the change README.md expression (build.yml)

Commit:
Automated publish: Mon Oct 10 17:39:34 UTC 2022 c2666e9fc2dd43a6bfee83883cd6e6c0fb0926e7

ayusavin committed:
Merge pull request #1 from Fantast03/feature-MoveCommand

Feature move command

Commit:
Automated publish: Sun Oct 16 10:57:38 UTC 2022 7f3826832f99d6d96559375db28fb0685468a3ae

ayusavin committed:
Rotate features

Added IRotatable, IAngle interfeaces, implemented Rotate command

Commit:
Automated publish: Thu Nov 17 12:08:53 UTC 2022 63dbd706dcc4fa8ea933dcd06039192ed58bf1b2

ayusavin committed:
Merge branch 'dev' into feature-RotateCommand

ayusavin committed:
Merge pull request #2 from Fantast03/feature-RotateCommand

Feature rotate command

Commit:
Automated publish: Thu Nov 17 12:26:40 UTC 2022 7934cf7cf9e58cd50deb2f39674165d2898f3920

ayusavin committed:
Update build.yml

Deleted pull_request trigger, added manual trigger

zaryanezrya commented on Dockerfile:
```diff
@@ -0,0 +1,15 @@
+FROM mcr.microsoft.com/dotnet/sdk:6.0-alpine as build
```
Убрать

ayusavin replied:
В Dockerfile были внесены стейджы с дев окружением, тестированием и генераций ковередж репортов, все предыдущее содержание было удалено за ненадобностью.

zaryanezrya commented on src/SpaceBattle/Entities/MoveCommand.cs:
```diff
@@ -0,0 +1,27 @@
+
+    public void Run()
+    {
+        if (movable == null || movable.Position == null || movable.MoveSpeed == null ||
```
Детали реализации сложения

ayusavin replied:
Операция сложения теперь инъектируется через IoC контейнер

ayusavin committed:
Added IoC container and Service Locator

Dependencies now resolve from Container, Move and Rotate commands addition operations implementations take out from Commands implementations files.

ayusavin committed:
Fixed accidentally corrupted workflow

ayusavin committed:
Released StartMoveCommand and tests

Added queue, uobject and MoveCommandStartable interfaces. Implemented StartMoveCommand command.

Commit:
Automated publish: Thu Dec 15 11:06:10 UTC 2022 2df2b823d63340d264515086cdce0e326d881726

ayusavin committed:
Merge pull request #4 from Fantast03/feature-StartMoveCommand

Feature start move command

ayusavin committed:
Refactoring of tests and format code

InjectContainer dependencies now in their own namespaces and directories. Added ReRegisterable thread non safe IoC for test purpouses

Commit:
Automated publish: Thu Dec 15 16:10:43 UTC 2022 590f3d4ed3702055108fde833b2ba52a7058eee1
